odilo.py

```diff
@@ -0,0 +1,581 @@
+            if content:
+                self.log.warn(content)
+            return None
+
+    @staticmethod
+    def _do_get(url, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
+
+        if 'allow_redirects' not in kwargs:
+            kwargs['allow_redirects'] = True
+
+        response = HTTP.get_with_timeout(url, headers=headers, **kwargs)
+        return response.status_code, response.headers, response.content
+
+    @staticmethod
+    def _do_post(url, payload, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
+
+        return HTTP.post_with_timeout(url, payload, headers=headers, **kwargs)
+
+
+class MockOdiloAPI(OdiloAPI):
+    @classmethod
+    def mock_collection(cls, _db):
+        library = DatabaseTest.make_default_library(_db)
+        collection, ignore = get_one_or_create(
+            _db, Collection,
+            name="Test Odilo Collection",
+            create_method_kwargs=dict(
+                external_account_id=u'library_id_123',
+            )
+        )
+        integration = collection.create_external_integration(
+            protocol=ExternalIntegration.ODILO
+        )
+        integration.password = u'abcdef123hijklm'
+        library.collections.append(collection)
+        return collection
+
+
+class OdiloRepresentationExtractor(object):
+    """Extract useful information from Odilo's JSON representations."""
+
+    log = logging.getLogger("OdiloRepresentationExtractor")
+
+    format_data_for_odilo_format = {
+        "ACSM": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM
+        ),
+        "PDF": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
+        "EPUB": (
+            Representation.EPUB_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
+        "MP3": (
+            Representation.MP3_MEDIA_TYPE, DeliveryMechanism.STREAMING_AUDIO_CONTENT_TYPE
+        ),
+        "MP4": (
+            Representation.MP4_MEDIA_TYPE, DeliveryMechanism.STREAMING_VIDEO_CONTENT_TYPE
+        ),
+        "WMV": (
+            Representation.WMV_MEDIA_TYPE, DeliveryMechanism.STREAMING_VIDEO_CONTENT_TYPE
+        ),
+        "JPG": (
+            Representation.JPEG_MEDIA_TYPE, DeliveryMechanism.NO_DRM
+        ),
+        "SCORM": (
+            Representation.ZIP_MEDIA_TYPE, DeliveryMechanism.NO_DRM
+        )
+    }
+
+    odilo_medium_to_simplified_medium = {
+        "ACSM": Edition.BOOK_MEDIUM,
+        "PDF": Edition.BOOK_MEDIUM,
+        "EPUB": Edition.BOOK_MEDIUM,
+        "MP3": Edition.AUDIO_MEDIUM,
+        "MP4": Edition.VIDEO_MEDIUM,
+        "WMV": Edition.VIDEO_MEDIUM,
+        "JPG": Edition.IMAGE_MEDIUM,
+        "SCORM": Edition.ELECTRONIC_FORMAT
```

Comment:
ELECTRONIC_FORMAT is used to distinguish between ebooks and print books. SCORM needs a new medium constant. I think COURSEWARE_MEDIUM would be most appropriate.

Reply:
Ok, I created that constant. Also I've checked where Edition.BOOK_MEDIUM is used within the Edition class, in order to add Edition.IMAGE_MEDIUM and the new Edition.COURSEWARE_MEDIUM.